src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -30,42 +36,20 @@ class TaskReplaceActor(
     val eventBus: EventStream,
     val readinessCheckExecutor: ReadinessCheckExecutor,
     val runSpec: RunSpec,
-    promise: Promise[Unit]) extends Actor with Stash with ReadinessBehavior with StrictLogging {
+    promise: Promise[Unit]) extends Actor with Stash with StrictLogging {
   import TaskReplaceActor._
 
-  // All existing instances of this app independent of the goal.
-  //
-  // Killed resident tasks are not expunged from the instances list. Ignore
-  // them. LaunchQueue takes care of launching instances against reservations
-  // first
-  val currentInstances = instanceTracker.specInstancesSync(runSpec.id)
-
-  // In case previous master was abdicated while the deployment was still running we might have
-  // already started some new tasks.
-  // All already started and active tasks are filtered while the rest is considered
-  private[this] val (instancesAlreadyStarted, oldInstances) = {
-    currentInstances.partition(_.runSpecVersion == runSpec.version)
-  }
+  val pathId: PathId = runSpec.id
 
-  // Old and new instances that have the Goal.Running
-  val activeInstances = currentInstances.filter(_.state.goal == Goal.Running)
+  // All running instances of this app
+  var instances: mutable.Map[Instance.Id, Instance] = instanceTracker.specInstancesSync(runSpec.id).map { i => i.instanceId -> i }(collection.breakOut)
```

Comment:
I'd go with either `val instances: mutable.Map` or `var instances: immutable.Map` but not both

Reply:
That's a good point. I am unsure yet how we are going to accumulate all instance states.